src/components/GlobalInfo.tsx

```diff
@@ -78,45 +80,47 @@ export function GlobalInfo() {
         <BoxTitle>{data.repo}</BoxTitle>
       </GlobalInfoEntry>
       <Spacer />
-      <div>
-        <GlobalInfoEntry>
-          <FontAwesomeIcon icon={branchIcon} color="#333" />
-          <Form method="post" action=".">
-            <SelectWithEllipsis
-              disabled={transitionState.state !== "idle"}
-              name="newBranch"
-              value={branch}
-              id="branch-selector"
-              onChange={(event) => {
-                const target = event.target
-                setBranch(event.target.value as string)
-                target.form?.submit()
-              }}
-            >
-              {Object.keys(data.refs.heads).map((key) => {
-                const hash = data.refs.heads[key]
-                const branchName = key
-                const hashShortened = hash.substring(0, 7)
-                return (
-                  <OptionWithEllipsis key={hash} value={branchName}>
-                    {branchName}
-                  </OptionWithEllipsis>
-                )
-              })}
-            </SelectWithEllipsis>
-          </Form>
-        </GlobalInfoEntry>
-        <Spacer />
-        <div>{!transitionState.submission?.formData.has("newBranch") ? "" : `Analyzing branch ${branch}`}</div>
-        <Spacer />
-        <strong>Analyzed: </strong>
-        <span>{dateTimeFormatShort(data.lastRunEpoch)}</span>
-        <Spacer />
-        <strong>As of commit: </strong>
-        <Code inline title={data.commit.message ?? "No commit message"}>
-          {data.commit.hash.slice(0, 7)}
-        </Code>
-      </div>
+      <GlobalInfoEntry>
+        <FontAwesomeIcon icon={branchIcon} color="#333" />
+        <StyledForm method="post" action=".">
+          <SelectWithEllipsis
+            disabled={transitionState.state !== "idle"}
+            name="newBranch"
+            value={branch}
+            id="branch-selector"
+            onChange={(event) => {
```

Comment:
You can get rid of the ref and utilize remix capabilities with the useSubmit hook like this:
https://remix.run/docs/en/v1.4.0-pre.2/api/remix#usesubmit

Reply:
Uh nice

Reply:
This should also prevent the refresh, since remix makes forms not refresh the entire page when we use their API's (e.g. submitting a remix Form element, or using the useSubmit hook)

Reply:
I've added useSubmit but couldn't figure out how to **not** use useRef

Reply:
I've pushed some changes that uses navigate instead for switching branching. This still leverages the cool remix capabilities.